lib/data/shared_preferences_helper.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+class SharedPreferencesHelper {
+  Future<SharedPreferences> get _getInstance => SharedPreferences.getInstance();
+
+  setUser(bool status) async {
+    SharedPreferences instance = await _getInstance;
+    await instance.setBool('USER', status);
+  }
+
+  getUser() async {
+    SharedPreferences instance = await _getInstance;
+    bool user = instance.getBool('USER');
```

Comment:
Essa variável pode ser nula, como, por exemplo, na primeira vez que o aplicativo for acessado por alguém. Sendo assim, faça essa alteração.